tests/semantic_analyser.cpp

```diff
@@ -2702,6 +2702,20 @@ TEST(semantic_analyser, string_size)
   ASSERT_EQ(var_assign->var->type.GetField(0).type.GetSize(), 6UL);
 }
 
+TEST(semantic_analyser, subprog_return)
+{
+  test("fn f(): void { return; }", 0);
+  test("fn f(): void { return 1; }", 1);
+  test("fn f(): int64 { return; }", 1);
+  test("fn f(): int64 { return 1; }", 0);
+}
+
+TEST(semantic_analyser, subprog_arguments)
+{
+  test("fn f(int64 $a): int64 { return $a; }", 0);
```

Comment:
Sorry if this was discussed elsewhere already, but the argument syntax feels a bit odd to me.

The function syntax is `fn <idents and stuff>: <ret type>`. It seems to me it would follow that argument syntax should also be `<ident>: <type>`, but here it's reversed.

IIRC we also added type annotations for local variables as `$var: type = <blah>` right?

Reply:
In that case I am in favor of `$arg: type` syntax

Reply:
I wonder if this type syntax can be expanded to other cases, like:

```
kprobe:some_kernel_fn(a: int, b: int): int
```

Basically adding the option to specify function signatures in places (althoug with btf and dwarf that might be a bit late). In that case, i wonder if having `->` like rust,python will make parsing easier